src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/courtsentencing/CourtSentencingService.kt

```diff
@@ -96,113 +97,181 @@ class CourtSentencingService(
     }
   }
 
-  private fun isDpsCreated(additionalInformation: AdditionalInformation) =
-    additionalInformation.source != CreatingSystem.NOMIS.name
+  private fun isDpsCreated(source: String) =
+    source != CreatingSystem.NOMIS.name
 
   suspend fun createCourtAppearance(createEvent: CourtAppearanceCreatedEvent) {
     val courtCaseId = createEvent.additionalInformation.courtCaseId
+    val source = createEvent.additionalInformation.source
     val courtAppearanceId = createEvent.additionalInformation.courtAppearanceId
     val offenderNo: String = createEvent.personReference.identifiers.first { it.type == "NOMS" }.value
     val telemetryMap = mutableMapOf(
       "dpsCourtCaseId" to courtCaseId,
       "dpsCourtAppearanceId" to courtAppearanceId,
       "offenderNo" to offenderNo,
     )
-    synchronise {
-      name = EntityType.COURT_APPEARANCE.displayName
-      telemetryClient = this@CourtSentencingService.telemetryClient
-      retryQueueService = courtSentencingRetryQueueService
-      eventTelemetry = telemetryMap
-
-      checkMappingDoesNotExist {
-        courtCaseMappingService.getMappingGivenCourtAppearanceIdOrNull(courtAppearanceId)
-      }
-
-      transform {
-        val courtCaseMapping = courtCaseMappingService.getMappingGivenCourtCaseIdOrNull(dpsCourtCaseId = courtCaseId)
-          ?: throw ParentEntityNotFoundRetry(
-            "Attempt to create a court appearance on a dps court case without a nomis mapping. Dps court case id: $courtCaseId not found for DPS court appearance $courtAppearanceId",
-          )
-
-        val courtAppearance = courtSentencingApiService.getCourtAppearance(courtAppearanceId)
+    if (isDpsCreated(source)) {
```

Comment:
I've added some missing event source checks which has complicated the PR somewhat